src/main/java/org/opensrp/service/PhysicalLocationService.java

```diff
@@ -485,4 +504,29 @@ public LocationTree buildLocationHierachyFromLocation(String locationId, boolean
 		return locationTree;
 	}
 
+	private void persistInRedisCache(List<AssignedLocations> assignedLocations, String username) {
+		List<Date> toDates = new ArrayList<>();
+		Boolean toDateExists = false;
+		for (AssignedLocations assignedLocation : assignedLocations) {
+			if (assignedLocation.getToDate() != null) {
+				toDates.add(assignedLocation.getToDate());
+				toDateExists = true;
+			}
+		}
+		long expiryTime = new Date().getTime() >= findMinimumDate(toDates).getTime() ?
+				new Date().getTime() - findMinimumDate(toDates).getTime() : findMinimumDate(toDates).getTime() - new Date().getTime();
+		expiryTime = expiryTime / 1000 ;
+		hashOps.put(username, ASSIGNED_LOCATIONS_HASH_KEY, assignedLocations);
```

Comment:
This computation should be done only when toDateExists

Reply:
Moved it inside if block